### ISSUE TITLE
KafkaConsumer on 'event.error' not catching errors

### ISSUE THREAD

**Issue by TechnoChimp** [Blizzard/node-rdkafka#844]
It seems the translation from a kafka error event to an event.error event is not happening.

My configuration:

```
const KafkaConsumerInstance: KafkaConsumer = new Kafka.KafkaConsumer.createReadStream({
    'group.id': config.kafka.groupID,
    'metadata.broker.list': config.kafka.brokers,
    'sasl.username': config.kafka.username,
    'sasl.password': config.kafka.password,
    'sasl.mechanisms': config.kafka.saslMechanisms,
    'security.protocol': 'sasl_ssl',
    'socket.keepalive.enable': true,
    'enable.auto.commit': false,
    
}, topicConfig, {'topics': [config.kafka.topic]})

KafkaConsumerInstance.on('event.error', function(err) {
    console.log('Got event.error event')
    if (err) {
        Logger.info(`Error: ${err}`, null, LogLevel.error)
    } 
})
```

When I introduce an error condition, neither log from this block is generated as we would expect. Instead I get the following indicating that the original 'error' event was never handled:

```
****path_here****/node_modules/node-rdkafka/lib/error.js:411
  return new LibrdKafkaError(e);
         ^
Error: Local: Broker transport failure
    at Function.createLibrdkafkaError [as create] (****path_here****/node_modules/node-rdkafka/lib/error.js:411:10)
    at ****path_here****/node_modules/node-rdkafka/lib/kafka-consumer.js:454:29
```

For testing purposes, I modified the typings in the node-rdkafka module to allow me to listen for 'error'. Then changing my code to listen for 'error', I was then able to catch and handle the errors as I would expect.

Additionally, changing all the instances of emit('error'...) in kafka-consumer-stream.js to emit('event.error'...) also fixes the problem, and listening to 'event.error' then works as expected. I'm guessing this is probably the fix.

**Comment by TechnoChimp** [comment]
Checking back to see if anyone has had a chance to check into this. It seems that it may be a pretty trivial fix, but is causing us a pretty big headache. (Uncaught errors are currently causing our services using this to crash)

**Comment by TechnoChimp** [comment]
Nevermind, I realize this was an issue with us not understanding the Consumer vs ConsumerStream objects. We had incorrectly typed our ConsumerStream as a KafkaConsumer: Issue number one. However, we also realized that the error event from the ConsumerStream can be listened for once we corrected our typing. 

This was the correct way:

const KafkaConsumerInstance: ConsumerStream = new Kafka.KafkaConsumer.createReadStream({...

KafkaConsumerInstance.on('error', function(err) {
    console.log('Got error event')
    if (err) {
        Logger.info(`Error: ${err}`, null, LogLevel.error)
    } 
})
